src/core/components/ChatInput.tsx

```diff
@@ -249,15 +252,7 @@ const ChatInput = ({ setShouldAutoScroll }: ChatInputProps) => {
       const totalFilesAfterUpdate =
         uploadedFiles.length + event.target.files.length;
       if (totalFilesAfterUpdate > maximumFileUploads) {
-        setUploadingState({
-          isActive: true,
-          isSupportedFile: false,
-          errorMessage: `Maximum ${maximumFileUploads} files allowed.`,
-        });
-
-        setTimeout(() => {
-          resetUploadState();
-        }, 2000);
+        setUploadError(`Maximum ${maximumFileUploads} files allowed.`);
```

Comment:
Now the view just needs to specify the error message because the function returned from the hook handles the two state updates with the set timeout.